### ISSUE TITLE
client:codegen failing when run with npx/yarn

### ISSUE THREAD

**Issue by lsanwick** [apollographql/apollo-tooling#881]
I'm seeing strange errors when attempting to run `apollo client:codegen` under yarn/npm instead of just running a globally installed version.

I've verified that the node version and `apollo` version are the same in both instances, see my shell log below:

It works successfully when I run it via a globally installed `apollo` (i.e. from `yarn global add apollo`)
```
$> apollo client:codegen __types__ --target=typescript
 ✔ Loading Apollo Project
 ✔ Generating query files with 'typescript' target - wrote 2 files
```

If I run the same command via `npx`, it fails.
```
$> npx apollo client:codegen __types__ --target=typescript
  ✔ Loading Apollo Project
  ✖ Generating query files with 'typescript' target
    → Cannot query field "user" on type "Query"
(node:32387) UnhandledPromiseRejectionWarning: GraphQLError: Cannot query field "user" on type "Query"
    at Compiler.compileSelection (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:119:27)
    at selections.selectionSetNode.selections.map.selectionNode (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:76)
    at Array.map (<anonymous>)
    at Compiler.compileSelectionSet (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:18)
    at Compiler.compileOperation (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:86:32)
    at Object.compileToIR (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:15:44)
    at Object.generate [as default] (/Users/<PROJECT>/node_modules/apollo/lib/generate.js:75:36)
    at write (/Users/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:67:42)
    at GraphQLClientProject.project.onDiagnostics [as _onDiagnostics] (/Users/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:91:17)
    at GraphQLClientProject.validate (/Users/<PROJECT>/node_modules/apollo-language-server/lib/project/client.js:102:18)
(node:32387) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:32387) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
GraphQLError: Cannot query field "user" on type "Query"
    at Compiler.compileSelection (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:119:27)
    at selections.selectionSetNode.selections.map.selectionNode (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:76)
    at Array.map (<anonymous>)
    at Compiler.compileSelectionSet (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:18)
    at Compiler.compileOperation (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:86:32)
    at Object.compileToIR (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:15:44)
    at Object.generate [as default] (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/generate.js:75:36)
    at write (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:67:42)
    at Task.task (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:93:36)
```

Same thing when using `yarn`.
```
$> yarn run apollo client:codegen __types__ --target=typescript
yarn run v1.12.3
$ /Users/<PROJECT>/node_modules/.bin/apollo client:codegen __types__ --target=typescript
  ✔ Loading Apollo Project
  ✖ Generating query files with 'typescript' target
    → Cannot query field "user" on type "Query"
(node:32548) UnhandledPromiseRejectionWarning: GraphQLError: Cannot query field "user" on type "Query"
    at Compiler.compileSelection (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:119:27)
    at selections.selectionSetNode.selections.map.selectionNode (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:76)
    at Array.map (<anonymous>)
    at Compiler.compileSelectionSet (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:18)
    at Compiler.compileOperation (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:86:32)
    at Object.compileToIR (/Users/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:15:44)
    at Object.generate [as default] (/Users/<PROJECT>/node_modules/apollo/lib/generate.js:75:36)
    at write (/Users/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:67:42)
    at GraphQLClientProject.project.onDiagnostics [as _onDiagnostics] (/Users/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:91:17)
    at GraphQLClientProject.validate (/Users/<PROJECT>/node_modules/apollo-language-server/lib/project/client.js:102:18)
(node:32548) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:32548) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
GraphQLError: Cannot query field "user" on type "Query"
    at Compiler.compileSelection (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:119:27)
    at selections.selectionSetNode.selections.map.selectionNode (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:76)
    at Array.map (<anonymous>)
    at Compiler.compileSelectionSet (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:106:18)
    at Compiler.compileOperation (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:86:32)
    at Object.compileToIR (~/Documents/repos/<PROJECT>/node_modules/apollo-codegen-core/lib/compiler/index.js:15:44)
    at Object.generate [as default] (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/generate.js:75:36)
    at write (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:67:42)
    at Task.task (~/Documents/repos/<PROJECT>/node_modules/apollo/lib/commands/client/codegen.js:93:36)
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

The versions appear identical in either mode, as far as I can tell.
```
$> apollo --version
apollo/2.1.9 darwin-x64 node-v10.14.1
$> yarn run apollo --version
yarn run v1.12.3
$ /Users/<PROJECT>/node_modules/.bin/apollo --version
apollo/2.1.9 darwin-x64 node-v10.14.1
✨  Done in 0.29s.
```

**Comment by Slessi** [comment]
A workaround for this problem in `2.3.0` is using either an absolute url or `../`

What worked in `2.1.8`:
- `./src/**/*.graphql`

What works in `2.3.0`:
- `../src/**/*.graphql`
- `/<full_project_dir>/talexio-hr-frontend/src/**/*.graphql`

**Comment by Slessi** [comment]
A workaround for this problem in `2.3.0` is using either an absolute url or `../` for the `includes` flag (or the deprecated `types`, I guess?`

What worked in `2.1.8`:
- `./src/**/*.graphql`

What works in `2.3.0`:
- `../src/**/*.graphql`
- `/<full_project_dir>/src/**/*.graphql`

**Comment by trevor-scheer** [comment]
Hey @lsanwick, thanks for submitting this issue. Would you mind letting me know if this has been resolved for you in the latest version of `apollo`? If not, please let me know the command and output, as well as anything you might have in an `apollo.config.js` file.

**Comment by chrisdrackett** [comment]
@trevor-scheer I can confirm that I'm getting the same behavior after updating to Apollo `2.12.3`. I have the following in my config:

```
module.exports = {
  client: {
    includes: ['../core/**/*.{js,ts,tsx}'],
    service: {
      name: 'server',
      localSchemaFile: './server/generated/schema.graphql',
    },
  },
}
```

and I'm getting a similar output:

```
 ▲ ~/src/calm yarn codegen                                                                                  13s  feature/auth :: 20h :: ● :: ⬡
yarn run v1.16.0
$ apollo client:codegen --watch --target typescript __types__
  ✔ Loading Apollo Project
  ✖ Generating query files with 'typescript' target
    → Cannot query field "tasks" on type "Query"
Press any key to stop.
(node:21500) UnhandledPromiseRejectionWarning: GraphQLError: Cannot query field "tasks" on type "Query"
    at Compiler.compileSelection (/Users/cwd/src/calm/node_modules/apollo-codegen-core/lib/compiler/index.js:119:27)
    at /Users/cwd/src/calm/node_modules/apollo-codegen-core/lib/compiler/index.js:106:76
    at Array.map (<anonymous>)
    at Compiler.compileSelectionSet (/Users/cwd/src/calm/node_modules/apollo-codegen-core/lib/compiler/index.js:106:18)
    at Compiler.compileOperation (/Users/cwd/src/calm/node_modules/apollo-codegen-core/lib/compiler/index.js:86:32)
    at Object.compileToIR (/Users/cwd/src/calm/node_modules/apollo-codegen-core/lib/compiler/index.js:15:44)
    at Object.generate [as default] (/Users/cwd/src/calm/node_modules/apollo/lib/generate.js:78:36)
    at write (/Users/cwd/src/calm/node_modules/apollo/lib/commands/client/codegen.js:67:54)
    at GraphQLClientProject._onDiagnostics (/Users/cwd/src/calm/node_modules/apollo/lib/commands/client/codegen.js:84:29)
    at GraphQLClientProject.validate (/Users/cwd/src/calm/node_modules/apollo-language-server/lib/project/client.js:132:18)
(node:21500) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:21500) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.```

**Comment by chrisdrackett** [comment]
so now based on https://github.com/apollographql/apollo-tooling/issues/1305 I'm adding 

```
  "resolutions": {
    "vscode-uri": "1.0.6"
  },
```

to make sure my copy of Apollo 2.8.3 works correctly. Has anyone found workarounds or fixes for this issue yet?

**Comment by JakeDawkins** [comment]
Hi 👋 Thanks for working on figuring this out. 

Can anyone here provide a minimal reproduction repo (or confirm it's fixed) for this with steps to reproduce? I'm having trouble reproducing this issue in my test projects. I'll temporarily close this until it's reproducible!

**Comment by Slessi** [comment]
@JakeDawkins https://github.com/Slessi/apollo-tooling-881-repro here you go

**Comment by JakeDawkins** [comment]
Thanks @Slessi! This is really strange. I'll try to look into it :)

**Comment by JakeDawkins** [comment]
@Slessi I found the issue you were running into specifically. I'm not sure if it resolves the other issues in this thread, but yours should be fixed with the merge of #1463
